jax/experimental/jax2tf/tests/call_tf_test.py

```diff
@@ -65,6 +66,34 @@ def _named_test(**kwargs):
 
 class CallTfTest(tf_test_util.JaxToTfTestCase):
 
+  @classmethod
+  def setUpClass(cls):
+    # Pick one device from each available platform
+    cls.jax_platforms = []
+    for backend in ["cpu", "gpu", "tpu"]:
+      try:
+        devices = jax.devices(backend)
+      except RuntimeError:
+        devices = []
+      if devices:
+        cls.jax_platforms.append(devices[0].platform)
+
+    # One TF device of each device_type
+    cls.tf_devices = []
+    for tf_device in (tf.config.list_logical_devices("TPU") +
+                      tf.config.list_logical_devices("GPU") +
+                      tf.config.list_logical_devices()):
```

Comment:
I think `tf.config.list_logical_devices()` will return *all* devices, including TPUs and GPUs, if available. Why isn't the last call with no `device_type=` sufficient?

Reply:
The point is that I want a preference order: TPUs if present, then GPUs if present, then CPUs. It seems that, unlike JAX, TF would return  the CPUs first when called without device_type.

Reply:
Good point, I copied this from another place where I wanted to prioritize TPU/GPUs.